src/System.Management.Automation/engine/runtime/CompiledScriptBlock.cs

```diff
@@ -193,15 +192,28 @@ private void ReallyCompile(bool optimize)
         private void PerformSecurityChecks()
         {
             var scriptBlockAst = Ast as ScriptBlockAst;
-            if (scriptBlockAst == null)
+            if (scriptBlockAst == null || _isProductCode == true)
             {
```

Comment:
@PaulHigin @TravisEz13 can you please weigh in here?

My main purpose here is to avoid the false positive result from the suspicious scan on `Utility.psd1`, which declares `Add-Type` in it. Due to this false positive scan result, powershell will attempt to log the script block when evaluating `Utility.psd1`, which will trigger the call to `IsProductSignature` that triggers the expensive catalog signature invocations.

Reply:
In fact, we come from one point -  `ScriptBlock.CreateDelayParsedScriptBlock("string", isProductCode: true)`.
I don't review in depth but I don't found the pattern "isProductCode: false". Perhaps we could implement separate code path.

Reply:
Cool, I will remove the check for `_isProductCode == true` here.
Just to bring it up that `IsProductCode` is used in `LogScriptBlockCreation`  to decide whether we need to log the script text. So an attacker can still change this property and make the malicious script not logged in the event log.

Reply:
The check is removed.